commit ad49d68fb5be38acd71453f8166ea4d57fb944c7
Author: Alexander Shishkin <alexander.shishkin@linux.intel.com>
Date:   Fri Sep 15 15:44:42 2017 +0300

    UBUNTU: SAUCE: PCI: Disable broken RTIT_BAR of Intel TH
    
    BugLink: http://bugs.launchpad.net/bugs/1715833
    
    On some intergrations of the Intel TH the reported size of RTIT_BAR
    doesn't match its actual size, which leads to overlaps with other
    devices' resources.
    
    For this reason, we need to disable the RTIT_BAR on Denverton where
    it would overlap with XHCI MMIO space and effectively kill usb dead.
    
    Signed-off-by: Alexander Shishkin <alexander.shishkin@linux.intel.com>
    Acked-by: Kamal Mostafa <kamal@canonical.com>
    Acked-by: Andy Whitcroft <apw@canonical.com>
    Acked-by: Thadeu Lima de Souza Cascardo <cascardo@canonical.com>
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 06554048798e..2f8a511b7d65 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -4785,6 +4785,27 @@ static void quirk_intel_no_flr(struct pci_dev *dev)
 DECLARE_PCI_FIXUP_EARLY(PCI_VENDOR_ID_INTEL, 0x1502, quirk_intel_no_flr);
 DECLARE_PCI_FIXUP_EARLY(PCI_VENDOR_ID_INTEL, 0x1503, quirk_intel_no_flr);
 
+static void quirk_intel_th_rtit_bar(struct pci_dev *dev)
+{
+	struct resource *r = &dev->resource[4];
+
+	/*
+	 * Hello, Denverton!
+	 * Denverton reports 2k of RTIT_BAR (resource 4), which can't be
+	 * right given the 16 threads. When Intel TH gets enabled, the
+	 * actual resource overlaps the XHCI MMIO space and causes it
+	 * to die.
+	 * We're not really using RTIT_BAR at all at the moment, so it's
+	 * a safe choice to disable this resource.
+	 */
+	if (r->end == r->start + 0x7ff) {
+		r->flags = 0;
+		r->start = 0;
+		r->end   = 0;
+	}
+}
+DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_INTEL, 0x19e1, quirk_intel_th_rtit_bar);
+
 static void quirk_no_ext_tags(struct pci_dev *pdev)
 {
 	struct pci_host_bridge *bridge = pci_find_host_bridge(pdev->bus);
